src/sentry/ingest/transaction_clusterer/tree.py

```diff
@@ -84,17 +86,41 @@ def add_input(self, transaction_names: Iterable[str]) -> None:
                 node = node.setdefault(part, Node())
 
     def get_rules(self) -> List[ReplacementRule]:
+        """Returns the computed rules.
+
+        If no rules have been computed, computes them. If you add input after
+        the rules have been computed, you must recompute them manually.
```

Comment:
Let's remove `get_rules` and only provide a `compute_rules` method, which ideally does not store any `_rules` on the instance. There is AFAIK no use case for getting rules multiple times.

Reply:
I agree with simplifying the two methods into one, but why do you think `_rules` is not helpful? I think it makes the API simpler to understand. Why not take advantage of the class to improve readability?

Reply:
Feel free to use it as a helper variable, I just would not return cached rules, exactly because of what you wrote here: 

> If you add input after
        the rules have been computed, you must recompute them manually.